File: backend/src/main/java/outcobra/server/model/ExamTask.java
package outcobra.server.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class ExamTask {
    @Id
    private Long id;

    @NotNull
    private String task;

    @NotNull
    @ManyToOne
    private Exam exam;

    @NotNull
    private boolean isFinished;
}

File: backend/src/main/kotlin/outcobra/server/model/ExamTask.kt
package outcobra.server.model

import outcobra.server.model.interfaces.ParentLinked
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.validation.constraints.NotNull

@Entity
class ExamTask(@NotNull var task: String = "",
               @NotNull @ManyToOne val exam: Exam? = null,
               @NotNull var isFinished: Boolean = false)
    : ParentLinked, AbstractEntity() {
    override val parent: ParentLinked?
        get() = exam
}
